src/aleph/vm/pool.py

```diff
@@ -170,10 +187,14 @@ def forget_vm(self, vm_hash: ItemHash) -> None:
             pass
 
     async def stop(self):
-        """Stop all VMs in the pool."""
-
+        """Stop ephemeral VMs in the pool."""
         # Stop executions in parallel:
-        await asyncio.gather(*(execution.stop() for vm_hash, execution in self.executions.items()))
+        await asyncio.gather(*(execution.stop() for vm_hash, execution in self.get_ephemeral_executions()))
+
+    def get_ephemeral_executions(self) -> Iterable[VmExecution]:
+        for _vm_hash, execution in self.executions.items():
+            if not execution.persistent and execution.is_running:
+                yield execution
```

Comment:
Would it make sense to use a generator comprehension here ?

```suggestion
def get_ephemeral_executions(self) -> Iterable[VmExecution]:
    return (execution for _vm_hash, execution in self.executions.items()
            if execution.is_running and not execution.persistent)
```